src/Rings/MPolyMap/flattenings.jl

```diff
@@ -0,0 +1,247 @@
+  return SU
+end
+
+### Canonical maps involved in flattening of towers of polynomial rings
+@attr function _flat_to_orig(
+    S::AbstractAlgebra.Generic.MPolyRing{RingElemType}
+  ) where {RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
+                                 <:MPolyQuoLocalizedRingElem
+                                }
+          }
+  _flatten(S) # Will set the attribute
+  return get_attribute(S, :_flat_to_orig)
+end
+
+@attr function _orig_to_flat(
+    S::AbstractAlgebra.Generic.MPolyRing{RingElemType}
+  ) where {RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
+                                 <:MPolyQuoLocalizedRingElem
+                                }
+          }
+  _flatten(S) # Will set the attribute
+  return get_attribute(S, :_orig_to_flat)
+end
+
+@attr function _base_ring_to_flat(
+    S::AbstractAlgebra.Generic.MPolyRing{RingElemType}
+  ) where {RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
+                                 <:MPolyQuoLocalizedRingElem
+                                }
+          }
+  _flatten(S) # Will set the attribute
+  return get_attribute(S, :_base_ring_to_flat)
+end
+
+### Computation of induced morphisms on flattened towers of polynomial rings
+@attr function _flatten(
+    f::MPolyAnyMap{AbstractAlgebra.Generic.MPolyRing{RingElemType}, 
+                   AbstractAlgebra.Generic.MPolyRing{RingElemType},
+                   Nothing
+                  }
+  ) where {RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
+                                 <:MPolyQuoLocalizedRingElem
+                                }
+          }
+  S = domain(f)
+  T = codomain(f)
+  RS = base_ring(domain(f))
+
+  S_flat = _flatten(S)
+  T_flat = _flatten(T)
+  imgs = _orig_to_flat(T).(f.(gens(S))) # The first half of the images
+  imgs = vcat(imgs, _base_ring_to_flat(T).(gens(RS))) # the ones from the base ring
+  return hom(S_flat, T_flat, imgs)
+end
+
+@attr function _flatten(
+    f::MPolyAnyMap{AbstractAlgebra.Generic.MPolyRing{RingElemType}, 
+                   AbstractAlgebra.Generic.MPolyRing{RingElemType}
+                   # Note the missing requirement here: It allows for a non-trivial coefficient map
+                  }
+  ) where {RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
+                                 <:MPolyQuoLocalizedRingElem
+                                }
+          }
+  S = domain(f)
+  T = codomain(f)
+  RS = base_ring(domain(f))
+
+  S_flat = _flatten(S)
+  T_flat = _flatten(T)
+  imgs = _orig_to_flat(T).(f.(gens(S))) # The first half of the images
+  imgs = vcat(imgs, _orig_to_flat(T).(T.(coefficient_map(f).(gens(RS))))) # the ones from the base ring
+  return hom(S_flat, T_flat, imgs)
+end
+
+### Deflecting some of the basic methods for ideals in towers of polynomial rings
+@attr function _flat_counterpart(
```

Comment:
Why is this not called `_flatten`?

Reply:
You're right. It could also just be another method of the same function.